Kernel/VirtIO/VirtIOQueue.cpp

```diff
@@ -103,6 +72,26 @@ void* VirtIOQueue::get_buffer(size_t* size)
     return token;
 }
 
+FlatPtr VirtIOQueue::get_buffer_paddr(size_t *size)
+{
+    if (!new_data_available()) {
+        *size = 0;
+        return 0;
+    }
+
+    full_memory_barrier();
+
+    auto descriptor_index = m_device->rings[m_used_tail % m_queue_size].index;
+    // TODO: Check the spec for this
+    *size = m_descriptors[descriptor_index].length;
+    auto paddr = m_descriptors[descriptor_index].address;
+
+    m_used_tail++;
+
+    pop_buffer(descriptor_index);
```

Comment:
why does getting the physical address of the buffer results in popping off one of the buffers?